docs/self-managed/setup/deploy/openshift/redhat-openshift.md

```diff
@@ -4,75 +4,370 @@ title: "Red Hat OpenShift"
+
+The presence of routes is rooted in their specification predating Ingress. The functionality of routes differs from Ingress; for example, unlike Ingress, routes don't allow multiple services to be linked to a single route or the use of paths.
+
+To use these routes for the Zeebe Gateway, configure this through Ingress as well.
+
+#### Setting Up the application domain for Camunda 8
+
+The route created by OpenShift will use a domain to provide access to the platform. By default, you can use the OpenShift applications domain, but any other domain supported by the router can also be used.
+
+To retrieve the OpenShift applications domain (used as an example here), run the following command:
+
+```bash
+export OPENSHIFT_APPS_DOMAIN=$(oc get ingresses.config.openshift.io cluster -o jsonpath='{.spec.domain}')
+```
+
+Next, define the route domain that will be used for the Camunda 8 deployment. For example:
+
+```bash
+export DOMAIN_NAME="camunda.$OPENSHIFT_APPS_DOMAIN"
+
+echo "Camunda 8 will be reachable from $DOMAIN_NAME"
+```
+
+If you choose to use a custom domain instead, ensure it is supported by your router configuration and replace the example domain with your desired domain. For more details on configuring custom domains in OpenShift, refer to the official [custom domain OpenShift documentation](https://docs.openshift.com/dedicated/applications/deployments/osd-config-custom-domains-applications.html).
+
+#### Checking if HTTP/2 is enabled
+
+As the Zeebe Gateway also uses `gRPC` (which relies on `HTTP/2`), [HTTP/2 Ingress Connectivity must be enabled](https://docs.openshift.com/container-platform/latest/networking/ingress-operator.html#nw-http2-haproxy_configuring-ingress).
+
+To check if HTTP/2 is already enabled on your OpenShift cluster, run the following command:
+
+```bash
+oc get ingresses.config/cluster -o json | jq '.metadata.annotations."ingress.operator.openshift.io/default-enable-http2"'
+```
+
+Alternatively, if you use a dedicated IngressController for the deployment:
+
+```bash
+# List your IngressControllers
+oc -n openshift-ingress-operator get ingresscontrollers
+
+# Replace <ingresscontroller_name> with your IngressController name
+oc -n openshift-ingress-operator get ingresscontrollers/<ingresscontroller_name> -o json | jq '.metadata.annotations."ingress.operator.openshift.io/default-enable-http2"'
+```
+
+- If the output is `"true"`, it means HTTP/2 is enabled.
+- If the output is `null` or empty, HTTP/2 is not enabled.
+
+<details>
+   <summary>Enable HTTP/2</summary>
+
+If HTTP/2 is not enabled, you can enable it by running the following command:
+
+**IngressController configuration:**
+
+```bash
+oc -n openshift-ingress-operator annotate ingresscontrollers/<ingresscontroller_name> ingress.operator.openshift.io/default-enable-http2=true
+```
+
+**Global cluster configuration:**
+
+```bash
+oc annotate ingresses.config/cluster ingress.operator.openshift.io/default-enable-http2=true
+```
```

Comment:
For this one I didn't have enough permissions 🤷 

> Error from server (Forbidden): admission webhook "ingress-config-validation.managed.openshift.io" denied the request: Only privileged service accounts may access

I went for the controller annotation in the end